index.js

```diff
@@ -7,256 +7,273 @@ var _ = require('lodash');
+     */
+  EchoClient.prototype.queryAnalytics = function queryAnalytics(
+    token,
+    queryOperator,
+    queryParams,
+    useCache,
+    callback
+  ) {
     if (!token) {
-        throw new Error('Missing Persona token');
+      throw new Error('Missing Persona token');
     }
 
     if (!queryOperator) {
-        throw new Error('Missing Analytics queryOperator');
+      throw new Error('Missing Analytics queryOperator');
     }
 
     var validOperators = ['hits', 'average', 'sum', 'max', 'funnel'];
 
     if (validOperators.indexOf(queryOperator) === -1) {
-        throw new Error('Invalid Analytics queryOperator');
+      throw new Error('Invalid Analytics queryOperator');
     }
 
     if (!queryParams) {
-        throw new Error('Missing Analytics queryParams');
+      throw new Error('Missing Analytics queryParams');
     }
 
-    var constructQueryStringResponse = this._queryStringParams(queryParams);
+    var constructQueryStringResponse = queryStringParams(queryParams);
 
     if (!_.isEmpty(constructQueryStringResponse.errors)) {
-        this.error('One or more invalid analytics queryParams where supplied: ' + constructQueryStringResponse.errors.join());
-        throw new Error('Invalid Analytics queryParams');
+      error(
+        'One or more invalid analytics queryParams where supplied',
+        { err: constructQueryStringResponse.errors.join() }
+      );
+
+      throw new Error('Invalid Analytics queryParams');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
-        method: 'GET',
-        headers: {
-            'Content-Type': 'application/json',
-            'Authorization': 'Bearer ' + token
-        }
+      url: config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + token
+      }
     };
 
     if (useCache === false) {
-        requestOptions.headers['cache-control'] = 'none';
+      requestOptions.headers['cache-control'] = 'none';
     }
 
-    this.debug(JSON.stringify(requestOptions));
+    debug('request options', requestOptions);
 
-    var _this = this;
+    request.get(requestOptions, function onResp(err, response, rawBody) {
+      var statusCode = response && response.statusCode
+        ? response.statusCode
+        : 0;
 
-    request.get(requestOptions, function(err, response, body) {
-        if (err) {
-            _this.error('[echoClient] queryAnalytics error: ' + JSON.stringify(err));
-            callback(err);
-        } else {
-            _this._parseJSON(body, callback);
-        }
+      if (err || parseInt(statusCode / 100) !== 2) {
+        error('[echoClient] queryAnalytics error', { err: err, body: rawBody, statusCode: statusCode });
+        callback(err || 'invalid status code: ' + statusCode);
```

Comment:
Is 'invalid status code' meaningful? It's like implying we've provided an invalid code.  Maybe "Error querying analytics: " + statusCode.  Admittedly I'd hope we normally get the true 'err' passed back to us but in scenarios we don't then maybe a clearer message is good.   Do we actually know what scenarios Echo will return a non 2xx status code but not trigger the error callback on request.get ?